common/src/main/java/tachyon/util/network/NetworkAddressUtils.java

```diff
@@ -48,6 +50,144 @@
   private NetworkAddressUtils() {}
 
   /**
+   * Different types of services that client uses to connect. These types also indicate the service
+   * bind address
+   */
+  public enum ServiceType {
+
+    /**
+     * Master RPC service (Thrift)
+     */
+    MASTER_RPC("Tachyon Master RPC service", Constants.MASTER_HOSTNAME, Constants.MASTER_BIND_HOST,
+        Constants.MASTER_PORT, Constants.DEFAULT_MASTER_PORT),
+
+    /**
+     * Master web service (Jetty)
+     */
+    MASTER_WEB("Tachyon Master Web service", Constants.MASTER_WEB_HOSTNAME,
+        Constants.MASTER_WEB_BIND_HOST, Constants.MASTER_WEB_PORT,
+        Constants.DEFAULT_MASTER_WEB_PORT),
+
+    /**
+     * Worker RPC service (Thrift)
+     */
+    WORKER_RPC("Tachyon Worker RPC service", Constants.WORKER_HOSTNAME, Constants.WORKER_BIND_HOST,
+        Constants.WORKER_PORT, Constants.DEFAULT_WORKER_PORT),
+
+    /**
+     * Worker data service (Netty)
+     */
+    WORKER_DATA("Tachyon Worker data service", Constants.WORKER_DATA_HOSTNAME,
+        Constants.WORKER_DATA_BIND_HOST, Constants.WORKER_DATA_PORT,
+        Constants.DEFAULT_WORKER_DATA_PORT),
+
+    /**
+     * Worker web service (Jetty)
+     */
+    WORKER_WEB("Tachyon Worker Web service", Constants.WORKER_WEB_HOSTNAME,
+        Constants.WORKER_WEB_BIND_HOST, Constants.WORKER_WEB_PORT,
+        Constants.DEFAULT_WORKER_WEB_PORT);
+
+    // service name
+    public final String mServiceName;
+
+    // the key of connect hostname
+    public final String mHostNameKey;
+
+    // the key of bind hostname
+    public final String mBindHostKey;
+
+    // the key of port
+    public final String mPortKey;
+
+    // default port number
+    public final int mDefaultPort;
+
+    ServiceType(String serviceName, String hostNameKey, String bindHostKey, String portKey,
+        int defaultPort) {
+      mServiceName = serviceName;
+      mHostNameKey = hostNameKey;
+      mBindHostKey = bindHostKey;
+      mPortKey = portKey;
+      mDefaultPort = defaultPort;
+    }
+  }
+
+  /**
+   * Provides an externally resolvable hostname for client to communicate with the service. If the
+   * hostname is not explicitly specified, Tachyon will try to use the bind host. If the bind host
+   * is wildcard, Tachyon will automatically determine an appropriate hostname from local machine.
+   *
+   * @param service Service type used to connect
+   * @param conf Tachyon configuration used to look up the host resolution timeout
+   * @return the externally resolvable hostname that the client can use to communicate with the
+   *         service.
+   */
+  public static String getConnectHost(ServiceType service, TachyonConf conf) {
```

Comment:
Could you enumerate the various possibilities here (in the javadoc) as well as in the configuration settings (ie. hostname = bindhost or hostname and then bindhost is wild card). Also in the case of a misconfiguration, could we catch it somewhere?